File: app/src/main/java/org/jellyfin/androidtv/ui/presentation/InfoCardPresenter.java
package org.jellyfin.androidtv.ui.presentation;

import android.view.View;
import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;

import org.jellyfin.androidtv.ui.card.MediaInfoCardView;
import org.jellyfin.apiclient.model.entities.MediaStream;

public class InfoCardPresenter extends Presenter {
    class ViewHolder extends Presenter.ViewHolder {
        private MediaInfoCardView mInfoCardView;

        public ViewHolder(View view) {
            super(view);
            mInfoCardView = (MediaInfoCardView) view;

        }

        public void setItem(MediaStream ms) {
            mInfoCardView.setMediaStream(ms);
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        MediaInfoCardView infoView = new MediaInfoCardView(parent.getContext());

        infoView.setFocusable(true);
        infoView.setFocusableInTouchMode(true);
        return new ViewHolder(infoView);
    }

    @Override
    public void onBindViewHolder(Presenter.ViewHolder viewHolder, Object item) {
        if (!(item instanceof MediaStream)) return;
        MediaStream mediaItem = (MediaStream) item;

        ViewHolder vh = (ViewHolder) viewHolder;

        vh.setItem(mediaItem);
    }

    @Override
    public void onUnbindViewHolder(Presenter.ViewHolder viewHolder) {
        //Log.d(TAG, "onUnbindViewHolder");
    }
}

File: app/src/main/java/org/jellyfin/androidtv/ui/presentation/InfoCardPresenter.kt
package org.jellyfin.androidtv.ui.presentation

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import org.jellyfin.androidtv.ui.card.MediaInfoCardView
import org.jellyfin.sdk.model.api.MediaStream

class InfoCardPresenter : Presenter() {
	class ViewHolder(
		private val mediaInfoCardView: MediaInfoCardView
	) : Presenter.ViewHolder(mediaInfoCardView) {
		fun setItem(mediaStream: MediaStream) = mediaInfoCardView.setMediaStream(mediaStream)
	}

	override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
		val view = MediaInfoCardView(parent.context).apply {
			isFocusable = true
			isFocusableInTouchMode = true
		}

		return ViewHolder(view)
	}

	override fun onBindViewHolder(viewHolder: Presenter.ViewHolder, item: Any?) {
		if (item !is MediaStream) return
		if (viewHolder !is ViewHolder) return

		viewHolder.setItem(item)
	}

	override fun onUnbindViewHolder(viewHolder: Presenter.ViewHolder) = Unit
}
